### ISSUE TITLE
[icetray] I3::dataio::open ignores zst files unless they end in .i3.zst (Trac #2407)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2399]
I3::data::open's compression type detection has gotten too clever by half. Instead of using gz/bz2/zst filters if a file name ends with the corresponding extension, it requires that the file end in .i3.gz, .i3.bz2, .i3.zst, etc. This breaks file staging, which uses os.path.splitext() to preserve only the primary file extension, e.g. mapping `gsiftp://gridftp.icecube.wisc.edu/data/sim/IceCube/2016/filtered/level2/neutrino-generator/21002/0004000-0004999/Level2_IC86.2016_NuMu.021002.004215.i3.zst` to `/mnt/glidein/execute.172.16.142.76-73/dir_1739/tmpxA7B47/Level2_IC86.2016_NuMu.021002.004215.i3_000000.zst`. Files opened for writing don't have this restriction.

Migrated from https://code.icecube.wisc.edu/ticket/2407
```json
{
    "status": "closed", 
    "changetime": "2020-06-24T12:31:42", 
    "description": "I3::data::open's compression type detection has gotten too clever by half. Instead of using gz/bz2/zst filters if a file name ends with the corresponding extension, it requires that the file end in .i3.gz, .i3.bz2, .i3.zst, etc. This breaks file staging, which uses os.path.splitext() to preserve only the primary file extension, e.g. mapping `gsiftp://gridftp.icecube.wisc.edu/data/sim/IceCube/2016/filtered/level2/neutrino-generator/21002/0004000-0004999/Level2_IC86.2016_NuMu.021002.004215.i3.zst` to `/mnt/glidein/execute.172.16.142.76-73/dir_1739/tmpxA7B47/Level2_IC86.2016_NuMu.021002.004215.i3_000000.zst`. Files opened for writing don't have this restriction.", 
    "reporter": "jvansanten", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1593001902142004", 
    "component": "combo core", 
    "summary": "[icetray] I3::dataio::open ignores zst files unless they end in .i3.zst", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2020-02-04T20:31:41", 
    "milestone": "Autumnal Equinox 2020", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20200204T23:15:22: 
* olivas changed owner from "" to "olivas"
* olivas changed milestone from "" to "Vernal Equinox 2020"
* olivas changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20200205T10:14:33: 
* jvansanten changed status from "accepted" to "closed"
* jvansanten commented:

> In changeset 178832:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="178832"
> Preserve compound extensions ending in .zst in stager
> 
> Effectively fixes #2399. I3::dataio::open() still requires the compound
> extension ".i3.zst" to enable zst stream decompression, but since neither
> GNU nor BSD tar have built-in zstd support, composable filters are not
> currently required.
> ```


* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20200624T12:31:42: 
* olivas changed milestone from "Vernal Equinox 2020" to "Autumnal Equinox 2020"
* olivas commented:

> Milestone Vernal Equinox 2020 deleted
